odd-collector/config_examples/ckan.yaml

```diff
@@ -1,10 +1,29 @@
+# You can store configuration in AWS SSM Parameter Store, if so - this block must be filled,
+# if not - you can skip it, and only configuration from file will be used.
+#
+# SSM Parameter Store naming example:
+# Lets asume, you are using the example below, and want to store default_pulling_interval secret,
+# in this case you have to create secret with:
+#   name: /odd/collector_config/collector_1/collector_settings/default_pulling_interval
+#   value: 60
+secrets_backend: "odd_collector_sdk.secrets.aws.ssm_parameter_store.SSMParameterStoreSecretsBackend"
```

Comment:
We shouldn't make people learn everything about internal sdk structure. Let's name it in more human way like:
"AWSSystemsManagerParameterStore" or "AWSSecretsManager". And after parsing the config yaml we can just map those name on a proper class that we have to use. Because right now we are sticking on not even
 implementation but the file location. So I would move this class in a different directory or just rename I would need to rewrite configuraion as well even if changes have nothing with config file.

Reply:
valid point, on the other hand, this way we can easily do mapping only for our backend “implementations”, since we know by which name which class we actually need to import. But in the extension we assume that this implementation may be third-party, and we may not know anything about it in advance. In this case, of course, the question is how to deliver the required package to the collector, and there are several options (none of them have been implemented yet), for example, to offer to expand the functionality of the collector by delivering zip archives to some pre-installed folder , or through Entry points of Python packages, and possibly through extensions. But I agree, since we still don’t have such methods now and nothing accidental can get inside the collector yet, we can do it through mappings. And if the time comes to give the opportunity to make extensions, we can work on this separately!

Reply:
@AndreyNenashev so what do you think in this way. We will simplify provider explanation and implement mappings. And about "custom" users backend we gonna deal later?

Reply:
Approach updated.